packages/node_modules/@node-red/runtime/lib/storage/localfilesystem/library.js

```diff
@@ -57,41 +57,17 @@ function getFileMeta(root,path) {
 function getFileBody(root,path) {
     var body = "";
     var fn = fspath.join(root,path);
-    var fd = fs.openSync(fn,"r");
-    var size = fs.fstatSync(fd).size;
-    var scanning = true;
-    var read = 0;
-    var length = 50;
-    var remaining = "";
-    var buffer = Buffer.alloc(length);
-    while(read < size) {
-        var thisRead = fs.readSync(fd,buffer,0,length);
-        if (scanning) {
-            var data = remaining+buffer.slice(0,thisRead).toString();
-            read += thisRead;
-            var parts = data.split("\n");
-            if (read < size) {
-                remaining = parts.splice(-1)[0];
-            } else {
-                remaining = "";
-            }
-            for (var i=0;i<parts.length;i+=1) {
-                if (! /^\/\/ \w+: /.test(parts[i])) {
-                    scanning = false;
-                    body += (body.length > 0?"\n":"")+parts[i];
-                }
-            }
-            if (!scanning) {
-                body += remaining;
-            }
-        } else {
-            read += thisRead;
-            body += buffer.slice(0,thisRead).toString();
+    var data = fs.readFileSync(fn,'utf8');
+    var parts = data.split("\n");
+
+    parts.forEach((part) => {
+        if (! /^\/\/ \w+: /.test(part)) {
```

Comment:
This will filter out any line in the file that starts with `// name: `. That is not the required behaviour.

A library file will contain zero or more lines at the start that begin `// name: `. They should be ignored. But as soon as it reads a line that doesn't start like that, then everything from then on must be included in the body.

For example, the following file:

```
// name: A test function
// outputs: 2
msg.payload = "hi";
// returns: a value
return msg;
```

should result in the body:
```
msg.payload = "hi";
// returns: a value
return msg;
```

Reply:
As a separate issue there will be an issue handling multi-byte chars in the meta data loading of the `getFileMeta` function. I believe @kazuhitoyokoi is going to look at that part of it?

Reply:
@knolleary Yes. I'm looking at the problem. Untournetely, I have holidays until the next Tuesday. 🙏 Therefore, I will submit the fix next week.